akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -30,10 +30,22 @@ abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S,
 
   protected def initialAttributes: Attributes = Attributes.none
 
-  final override lazy val traversalBuilder: TraversalBuilder = {
-    val attr = initialAttributes
-    TraversalBuilder.atomic(GraphStageModule(shape, attr, this), attr)
-  }
+  private[this] final var _traversalBuilder: TraversalBuilder = _
```

Comment:
we can avoid the `lazy val` I think, please sanity check that assumption though?